nsqd/diskqueue.go

```diff
@@ -261,6 +261,14 @@ func (d *diskQueue) readOne() ([]byte, error) {
 		return nil, err
 	}
 
+	if msgSize < minMsgSize {
```

Comment:
I only think `DiskQueue` needs to protect itself from an undesirable `panic`, but perhaps we're just settling on the fact that we _do_ actually need to pass in a `min` and `max` in the `DiskQueue` constructor (so it can internally apply both).

That seems like the right happy medium to both our points and would be objectively better than the current implementation.  It would still not protect from actual message body corruption, however that would require more invasive changes (CRC).